borg/key.py

```diff
@@ -396,25 +400,46 @@ class KeyfileKey(KeyfileKeyBase):
     TYPE = 0x00
     FILE_ID = 'BORG_KEY'
 
+    def sanity_check(self, filename, id):
+        with open(filename, 'r') as fd:
+            line = fd.readline().strip()
+            if not line.startswith(self.FILE_ID):
+                raise KeyfileInvalidError(self.repository._location.canonical_path(), filename)
+            elif line[len(self.FILE_ID) + 1:] != id:
+                return None
+            else:
+                return filename
```

Comment:
this is now a mix of raising exceptions and signalling success via return value.
the success return value is no really  needed as it is the same value as given to this method. instead of returning None it could raise another exception.

in general exceptions could be catched by caller or not, depending on the requirements.